crates/generated_parser/src/early_errors.rs

```diff
@@ -909,6 +971,80 @@ impl VarEarlyErrorsContext for LexicalForBodyEarlyErrorsContext {
     }
 }
 
+// ===========================================================================
+// LabelledStatements
+// https://tc39.es/ecma262/#sec-labelled-statements
+// ===========================================================================
+
+pub struct LabelledStatementEarlyErrorsContext {
+    name: SourceAtomSetIndex,
+    is_loop: bool,
+}
+
+impl LabelledStatementEarlyErrorsContext {
+    pub fn new(name: SourceAtomSetIndex, is_loop: bool) -> Self {
+        Self { name, is_loop }
+    }
+
+    pub fn check_duplicate_label<'alloc>(
+        &self,
+        name: SourceAtomSetIndex,
+    ) -> EarlyErrorsResult<'alloc> {
+        // Static Semantics: ContainsDuplicateLabels
+        // https://tc39.es/ecma262/#sec-labelled-statements-static-semantics-containsduplicatelabels
+        //
+        //  LabelledStatement : LabelIdentifier : LabelledItem
+        //
+        // Static Semantics: Early Errors
+        // https://tc39.es/ecma262/#sec-scripts-static-semantics-early-errors
+        //
+        // * It is a Syntax Error if ContainsDuplicateLabels of StatementList with argument « » is
+        //   true.
+        //
+        // and
+        //
+        // https://tc39.es/ecma262/#sec-module-semantics-static-semantics-early-errors
+        //
+        // * It is a Syntax Error if ContainsDuplicateLabels of ModuleItemList with argument « » is
+        //   true.
+        //
+        // and
+        //
+        // https://tc39.es/ecma262/#sec-function-definitions-static-semantics-early-errors
+        // * It is a Syntax Error if ContainsDuplicateLabels of FunctionStatementList with argument
+        //   « » is true.
+        if name == self.name {
+            return Err(ParseError::DuplicateLabel);
+        }
+        Ok(())
+    }
+
+    pub fn on_unhandled_continue<'alloc>(&self, info: &ControlInfo) -> EarlyErrorsResult<'alloc> {
+        //  Continues outside of iterators and Unlabelled breaks can not be detected at the
+        //  function / script level easily, because we only have binding information there, not
+        //  whether the label in question is associated with a loop or not. The nesting errors
+        //  also cannot be detected at the {Break,Continue}Statement level, as we don't have that
+        //  information there either. So we are handling these errors here. This handles labelled
+        //  continues that would otherwise pass.
+        //
+        //  Static Semantics: Early Errors
+        //  https://tc39.es/ecma262/#sec-continue-statement-static-semantics-early-e
+        //
+        //  ContinueStatement : continue LabelIdentifier ;
+        //
+        //  * It is a Syntax Error if this ContinueStatement is not nested, directly
+        //    indirectly (but not crossing function boundaries), within an
+        //    IterationStatement.
+        if let Some(name) = info.label {
+            if !self.is_loop && info.kind == ControlKind::Continue && name == self.name {
```

Comment:
sorry, if `info.kind == ControlKind::Continue` is checked here, the method name shouldn't be `on_unhandled_continue`.
(because it means `info` can be break)

also, this seems not to be checking unhandled things, given this isn't top level.

then, `self.is_loop` should be checked at the beginning, maybe outside of the loop in the caller.
given it's constant, and the error checking is only about labelled statement with non-loop.

how about renaming this to `check_labelled_continue` or `check_labelled_continue_to_non_loop` or something?